src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs

```diff
@@ -600,6 +601,9 @@ private static HttpContext CloneHttpContext(HttpContext context)
             var newHttpContext = new DefaultHttpContext(features);
             newHttpContext.TraceIdentifier = context.TraceIdentifier;
 
+            var endpointFeature = context.Features.Get<IEndpointFeature>();
+            newHttpContext.SetEndpoint(endpointFeature?.Endpoint);
```

Comment:
@JamesNK is this a deep enough copy? Does the endpoint get disposed or reused anywhere?

Reply:
Endpoints are 🔥 immutable 🔥